[SO-1440] Attack Surface (#251)
cobalthq/cobalt-api-docs: versions/v2/content/attack-surface.md

---
weight: 14
title: "Attack Surface"
---

# Attack Surface

## Get Attack Surface Stats

```sh
curl --location 'https://api.us.cobalt.io/attack_surface/stats' \
  -H "Accept: application/vnd.cobalt.v2+json" \
  -H "Authorization: Bearer YOUR-PERSONAL-API-TOKEN" \
  -H "X-Org-Token: YOUR-V2-ORGANIZATION-TOKEN"

>   --data '{}'

Reviewer: Did you check if it works like this? According to the [swagger docs](https://api.us.co27.io/cobalt-api-docs/webjars/swagger-ui/index.html#/Attack%20Surface/getV2AttackSurfaceStats), also X-Reqeuest-Id is required.

Author: After myself and @roger-cobalt checked swagger and tested against https://api.roger.us.co27.io/attack_surface/stats we found that `X-Reqeuest-Id ` is not required for the GET request. 

It also not mentioned in the public API. Probably because it is auto generated if not passed in the headers

```

> The above command returns JSON structured like this:

```json

> {
>       "resource": {
>         "domains": "3",
>         "seen_hosts": "200",
>         "new_seen_hosts": "10"
>       }
> }

Reviewer: The links part is missing, see https://api.us.co27.io/cobalt-api-docs/webjars/swagger-ui/index.html#/Attack%20Surface/getV2AttackSurfaceStats
```suggestion
{
  "resource": {
    "domains": "3",
    "seen_hosts": "200",
    "new_seen_hosts": "10"
  },
  "links": {
    "ui": {
      "url": "https://api.cobalt.io/somelink"
    }
  }
}
```

```

This endpoint retrieves stats from Attack Surface for the organization

### HTTP Request

`GET https://api.us.cobalt.io/attack_surface/stats`

### Response Fields

> | Field           | Description                                                                         |

Reviewer: I guess also in this table the link to attack-surface in the platform should be mentioned.

Reviewer: The table also has a pretty random amount of whitespaces 🤔

Author: I copied this table from `versions/v2/content/dast-targets.md`

I thought the whitespaces might have served a purpose in the markdown but they don't so I updated it 👍 

> I guess also in this table the link to attack-surface in the platform should be mentioned.

By `link` you mean the parameter that was missing which has the link to the UI? If so I have added that missing piece 👍  If you mean something else let me know what I might have misunderstood

Reviewer: yes, the one to UI as mentioned also above on the example

---
weight: 14
title: "Attack Surface"
---

# Attack Surface

## Get Attack Surface

```sh
curl --location 'https://api.us.cobalt.io/attack_surface/stats' \
  -H "Accept: application/vnd.cobalt.v2+json" \
  -H "Authorization: Bearer YOUR-PERSONAL-API-TOKEN" \
  -H "X-Org-Token: YOUR-V2-ORGANIZATION-TOKEN"
```

> The above command returns JSON structured like this:

```json
{
  "resource": {
    "domains": "3",
    "seen_hosts": "200",
    "new_seen_hosts": "10"
  },
  "links": {
    "ui": {
      "url": "https://api.cobalt.io/links/base64-link"
    }
  }
}
```

> This endpoint retrieves stats from Attack Surface for the organization

Reviewer 2: ```suggestion
This endpoint retrieves stats from the attack surface that belong to the organization specified in the `X-Org-Token` header.
```

I apologize for the numerous change requests, but the information from the existing endpoint is truly superior.

Author: No problem, good suggestion

### HTTP Request

`GET https://api.us.cobalt.io/attack_surface/stats`

### Response Fields

> | Field            | Description                                                              |
> | ---------------- | ------------------------------------------------------------------------ |
> | `domains`        | Total amount of domains                                                  |
> | `seen_hosts`     | Amount of hosts with at least one IP found within the last 7 days        |
> | `new_seen_hosts` | Amount of hosts we found an IP for the first time within the last 7 days |
> | `links.ui.url`   | Contains URL to the attack surface of your Org                           |

Reviewer 2: ```suggestion
| Field            | Description                                                                                                   |
| ---------------- | ------------------------------------------------------------------------------------------------------------- |
| `domains`        | Total amount of domains                                                                                       |
| `seen_hosts`     | Amount of hosts with at least one IP found within the last 7 days                                             |
| `new_seen_hosts` | Amount of hosts we found an IP for the first time within the last 7 days                                      |
| `links.ui.url`   | A link to redirect an authorized user to the attack surface of the organization in the Cobalt web application |

```

What do you think?

Author: Nice! Strange that `npx prettier` gave you a different result than mine

---
weight: 14
title: "Attack Surface"
---

# Attack Surface

> ## Get Attack Surface Stats

Author: Removed the `--data '{}'` 👍

```sh
curl --location 'https://api.us.cobalt.io/attack_surface/stats' \
  -H "Accept: application/vnd.cobalt.v2+json" \
  -H "Authorization: Bearer YOUR-PERSONAL-API-TOKEN" \
  -H "X-Org-Token: YOUR-V2-ORGANIZATION-TOKEN"
  --data '{}'
```

> The above command returns JSON structured like this:

```json
{
      "resource": {
        "domains": "3",
        "seen_hosts": "200",
        "new_seen_hosts": "10"
      },
      "links": {
        "ui": {
          "url": "https://api.cobalt.io/links/base64-link"
        }
      }
}
```

This endpoint retrieves stats from Attack Surface for the organization

### HTTP Request

`GET https://api.us.cobalt.io/attack_surface/stats`

### Response Fields

| Field           | Description                                                                         |
|-----------------|-------------------------------------------------------------------------------------|
| `domains`      |Total amount of domains|
| `seen_hosts`    |Amount of hosts with at least one IP found within the last 7 days|
| `new_seen_hosts`     |Amount of hosts we found an IP for the first time within the last 7 days|

> | `links` |Contains URL to the attack surface of your Org|

Reviewer 2: ```suggestion
| `links` |Contains URLs to the attack surface of your Org|
```

?

Reviewer: It's just one URL in there, so we intentionally used singular.

Reviewer 2: Thanks for the clarification! The pentest and findings pages clarify this property better.

https://github.com/cobalthq/cobalt-api-docs/blob/4655cc256bf0c03dc6e8dfc4c5242bfc235492f3/versions/v2/content/pentests.md?plain=1#L95

https://github.com/cobalthq/cobalt-api-docs/blob/4655cc256bf0c03dc6e8dfc4c5242bfc235492f3/versions/v2/content/findings.md?plain=1#L182

I recommend aligning new docs to the existing ones.

💄 Please use a tool or CLI anything to format the markdown table. Here is one solution: `npx prettier—-write ./path/to.md`. Thanks!
